docs/src/types/vectors.md

```diff
@@ -0,0 +1,183 @@
+# Vectors
+
+## Passing in vectors
+You can pass a Rust std::vec::Vec into your contract function transparently. The following code calls a sway function which accepts and then returns a `Vec<SomeStruct<u32>>`.
+
+```rust,ignore
+{{#include ../../../packages/fuels/tests/harness.rs:passing_in_vec}}
+```
+
+You can use a vector just like you would use any other type -- i.e. a `[Vec<u32>; 2]` or a `SomeStruct<Vec<Bits256>>` etc.
+
+## Returning vectors
+There is a mandatory extra step to returning vectors -- you need to `log` every element before returning the vector.
+
+You must not mix any unrelated `log`s once you start logging vector elements. 
+
+These logs must be the last ones you make.
+
+
+### A simple case:
+```Rust
+contract;
+
+use std::logging::log;
+use std::vec::Vec;
+use std::option::Option;
+use std::assert::assert;
```

Comment:
```suggestion
use std::logging::log;
```
The other are now in the prelude. Assuming you're on `v0.24.4` or after.

Reply:
0.24.4 didn't want to compile.

I've just tried 0.24.5 and it works. Will update the CI version of forc and remove this.

Reply:
Ahhhh it is possible that `0.24.4` has `Vec` and `assert` only. `Option` needs `0.24.5`.